### PR TITLE
 Adding unit tests to OAuthApplicationMgtListener 

### PR DESCRIPTION
**Pull request by emswbandara**

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+                {true, true, OAUTH2, "otherPropName"},
+                {false, false, null, null}
+        };
+    }
+
+    @Test
+    public void testGetDefaultOrderId() {
+
+        int result = oAuthApplicationMgtListener.getDefaultOrderId();
+        assertEquals(result, 11, "Failed to get default order ID.");
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java]: Is the fail message correct?

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+        mockStatic(AuthorizationGrantCache.class);
+        when(AuthorizationGrantCache.getInstance()).thenReturn(mockAuthorizationGrantCache);
+
+        mockStatic(OAuthCache.class);
+        when(OAuthCache.getInstance()).thenReturn(mockOauthCache);
+    }
+
+    @DataProvider(name = "GetSPConfigData")
+    public Object[][] SPConfigData() {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java]: Shall we move the data provider just above the method it provides data to?

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+            ServiceProvider serviceProvider = createServiceProvider(1, true, true, OAUTH2, OAUTH_CONSUMER_SECRET);
+            Boolean result = oAuthApplicationMgtListener.doPostCreateApplication(serviceProvider, tenantDomain, userName);
+            assertTrue(result, "Post-create application failed.");
+        }
+    }
+
+    @DataProvider(name = "GetPostUpdateApplicationData")
+    public Object[][] PostUpdateApplicationData() {
+
+        return new Object[][]{
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java]: Having comments to relate the data provided with how it relates to the tested flow would make the test much readable and easy to debug/refactor/fix the unit tests. WDYT?

Comment (emswbandara) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java]: updated Data provider with comments on data entries.

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+                // was enabled before and disabled with application update, saas-tokens should be revoked.
+                {true, true, OAUTH2, OAUTH_CONSUMER_SECRET, true, true},
+                //test the normal flow of an oauth application when cache disabled and saas not enabled before.
+                {true, true, OAUTH, OAUTH_CONSUMER_SECRET, false, false},
+                //test addClientSecret() and updateAuthApplication() for other authentication types.
+                {true, true, "otherAuthType", "otherPropName", false, false},
+                //test addClientSecret() and for oauth applications with inboundRequestConfig properties without
+                // oauthConsumerSecret property.
+                {true, true, OAUTH2, "otherPropName", false, false},
+                //test addClientSecret() and updateAuthApplication() for the scenario where inboundAuthenticationConfig
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java]: Leave a space after // and start the comment with a Capital letter

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+    }
+
+    @Test
+    public void testDoPostGetApplicationExcludingFileBasedSPs() throws Exception {
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+
+            ServiceProvider serviceProvider = createServiceProvider(1, true, true, OAUTH2, OAUTH_CONSUMER_SECRET);
+            Boolean result = oAuthApplicationMgtListener.doPostGetApplicationExcludingFileBasedSPs(serviceProvider,
```
[the remark this answers is not in the document]
Comment (emswbandara) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java]: Fixed with #780

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+    public void doPreDeleteApplication() throws Exception {
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+
+            ServiceProvider serviceProvider = createServiceProvider(1, false, false, "otherAuthType",
+                    OAUTH_CONSUMER_SECRET);
+            when(mockAppMgtService.getApplicationExcludingFileBasedSPs(anyString(), anyString())).thenReturn(serviceProvider);
+
+            Boolean result = oAuthApplicationMgtListener.doPreDeleteApplication(spName, tenantDomain, userName);
```
[the remark this answers is not in the document]
Comment (emswbandara) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListenerTest.java]: Fixed with #780

**Pull request merged**